app/serializers/concerns/filter_has_many.rb

```diff
@@ -0,0 +1,31 @@
+module FilterHasMany
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def page(params = {}, scope = nil, context = {})
+      filters = has_many_filterable_by.map do |filter|
+        filter << params.delete(filter[1])
+      end.delete_if do |filter|
+        filter[2].nil?
+      end
+
+      scope = filters.reduce(scope || self.model_class.all) do |query, filter|
+        scope.joins(filter[0]).where(filter[0] => {id: filter[2]})
```

Comment:
Yes, but I'm passing the scope into the reduce block as the `query` variable. How it works now will only allow filtering by one has many param (which is as many as any of the models have).